### PR TITLE
Another batch of assorted test tweaks

### PR DESCRIPTION
**Pull request by mrc0mmand**
This adds a couple of sanity test for the newly introduced `ASSERT_*()` macros (along with one bugfix) and addresses several issues I gathered from CI logs in the past two weeks.

### CODE REVIEW

File: src/shared/tests.h [hunk 1]
```diff
@@ -251,7 +255,7 @@ static inline int run_test_table(void) {
         ({                                                                                                      \
                 const char* _expr1 = (expr1);                                                                   \
                 const char* _expr2 = (expr2);                                                                   \
```
Comment (YHNdnzj) [inline src/shared/tests.h]: Hmm, merge the `const char *` part while at it?

File: src/shared/tests.h [hunk 2]
```diff
@@ -251,7 +255,7 @@ static inline int run_test_table(void) {
...
-                if (strcmp(_expr1, _expr2) != 0) {                                                              \
+                if (!streq_ptr(_expr1, _expr2) != 0) {                                                          \
```
Comment (YHNdnzj) [inline src/shared/tests.h]:
```suggestion
                if (!streq_ptr(_expr1, _expr2)) {                                                          \
```

Comment (mrc0mmand) [reply src/shared/tests.h]: I have no idea how that managed to sneak in :)

File: src/test/test-macro.c [hunk 3]
```diff
@@ -1105,4 +1105,61 @@ TEST(u64_multiply_safe) {
         assert_se(u64_multiply_safe(UINT64_MAX, UINT64_MAX) == 0);
 }
 
+TEST(ASSERT) {
```
Comment (daandemeyer) [inline src/test/test-macro.c:1108]: _We have to go deeper_

**Review by yuwata — APPROVED** [review]

**Pull request merged**